### PR TITLE
[community][embeddings] infinity embedding local option

### PR DESCRIPTION
**Pull request by michaelfeil**
Ready for review - **drop-in-replacement for sentence-transformers inference.**

https://github.com/langchain-ai/langchain/discussions/17670

tldr from the discussion above -> around a 4x-22x speedup over using SentenceTransformers / huggingface embeddings. For more info: https://github.com/michaelfeil/infinity (pure-python dependency)

Checklist:

- [ x] PR message: **Delete this entire template message** and replace it with the following bulleted list
    - **Description:** a description of the change
    - **Issue:** [the issue # it fixes, if applicable](https://github.com/langchain-ai/langchain/discussions/17670)
    - **Dependencies:** infinity_emb, will raise a soft runtime error if N/A
    - **Twitter handle:** No twitter.
- [x] Pass lint and test: DONE that
- [x] Add tests and docs: If you're adding a new integration, please include 
  1. a test for the integration, preferably unit tests that do not rely on network access, I ADDED A UNIT TEST
  2. an example notebook showing its use. It lives in `docs/docs/integrations` directory. I ADDED and tested a notebook.

Additional guidelines:
- Make sure optional dependencies are imported within a function.
- Please do not add dependencies to pyproject.toml files (even optional ones) unless they are required for unit tests.
- Most PRs should not touch more than one package.
- Changes should be backwards compatible.
- If you are adding something to community, do not re-import it in langchain.

If no one reviews your PR within a few days, please @-mention one of baskaryan, efriis, eyurtsev, hwchase17.

### CODE REVIEW

**Comment by michaelfeil** [comment]
Finally made it past the CI again!

**Comment by michaelfeil** [comment]
@baskaryan Looking forward to your review again.

File: docs/docs/integrations/text_embedding/infinity_local.ipynb
Comment (baskaryan) [inline docs/docs/integrations/text_embedding/infinity_local.ipynb:1]: would it make sense to put this in the existing Infinity notebook, may be easier to find that way?

**Review by baskaryan — APPROVED** [review]
one comment, otherwise lgtm!

File: docs/docs/integrations/text_embedding/infinity_local.ipynb
Comment (michaelfeil) [reply docs/docs/integrations/text_embedding/infinity_local.ipynb:1]: Actually, since they are so different, I prefer them in two notebooks. Is it possible with keeping them separate?

Comment (michaelfeil) [reply docs/docs/integrations/text_embedding/infinity_local.ipynb:1]: Okay, consolidated into one notebook

Comment (michaelfeil) [reply docs/docs/integrations/text_embedding/infinity_local.ipynb:1]: @baskaryan this is done, thanks ;)

**Comment by michaelfeil** [comment]
@baskaryan Whats up with the vecel deployment, cant see this part of the CI, but it also should not be related to this PR (At most updated the notebook)

**Review by efriis — APPROVED** [review]
Notebook by mistake had some python code in a markdown cell! Caused docusaurus to read the `import numpy as np` statement as a wrong javascript import (because docusaurus is mdx)

Should be fixed now

**Comment by michaelfeil** [comment]
@efriis Thanks for the fix - that was an obvious one. Saw you're also biking in SF - feel free to message me at michaelfeil.eu, got to train for a half iron later in August.
